### PR TITLE
Added persistency of WellKnownConfig/OperatorDiscoveryResult

### PR DESCRIPTION
**Pull request by gorandostanic**

### CODE REVIEW

File: connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java [hunk 1]
```diff
@@ -29,18 +31,29 @@
...
+    private LastSeenConfigStore lastSeenStore;
 
     public MobileConnectSdkProfile(
             Context context,
             final OperatorDiscoveryConfig operatorDiscoveryConfig,
             boolean confidentialClient) {
         super(context, confidentialClient);
         this.operatorDiscoveryConfig = operatorDiscoveryConfig;
+
+        this.lastSeenStore = new LastSeenConfigStore();
```
Comment (atanasdb) [inline connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java]: move the init to the definition and make it final

File: connect/src/com/telenor/connect/AbstractSdkProfile.java [hunk 2]
```diff
@@ -20,12 +25,19 @@
...
     private volatile boolean isInitialized = false;
+    private LastSeenConfigStore lastSeenStore;
 
     public AbstractSdkProfile(
             Context context,
             boolean confidentialClient) {
         this.context = context;
         this.confidentialClient = confidentialClient;
+
+        this.lastSeenStore = new LastSeenConfigStore();
```
Comment (atanasdb) [inline connect/src/com/telenor/connect/AbstractSdkProfile.java]: move the init to the definition and make it final

File: connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java [hunk 3]
```diff
@@ -247,4 +271,29 @@ public void getUserInfo(
                     userInfoCallback);
         }
     }
+
+    private class LastSeenConfigStore {
```
Comment (atanasdb) [inline connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java]: Why don't you prefer the TokenStore style?
E.g. rename this class to OperatorDiscoveryStore, and the methods set() and get()
Same for Wellknown config class.

File: connect/src/com/telenor/connect/AbstractSdkProfile.java [hunk 4]
```diff
@@ -89,4 +108,29 @@ public void failure(RetrofitError error) {
                     }
                 });
     }
+
+    private class LastSeenConfigStore {
```
Comment (atanasdb) [inline connect/src/com/telenor/connect/AbstractSdkProfile.java]: Why don't you prefer the TokenStore style?

File: connect/src/com/telenor/connect/AbstractSdkProfile.java [hunk 5]
```diff
@@ -20,12 +25,19 @@
...
+    private LastSeenConfigStore lastSeenStore;
 
     public AbstractSdkProfile(
             Context context,
             boolean confidentialClient) {
         this.context = context;
         this.confidentialClient = confidentialClient;
+
+        this.lastSeenStore = new LastSeenConfigStore();
+        WellKnownAPI.WellKnownConfig lastSeen = lastSeenStore.getWellKnownConfig();
```
Comment (atanasdb) [inline connect/src/com/telenor/connect/AbstractSdkProfile.java]: why not just:
wellKnownConfig = lastSeenStore.getWellKnownConfig();

File: connect/src/com/telenor/connect/AbstractSdkProfile.java [hunk 6]
```diff
@@ -2,8 +2,11 @@
 
 import android.content.Context;
 
+import com.google.gson.Gson;
 import com.telenor.connect.id.ConnectIdService;
 import com.telenor.connect.utils.RestHelper;
+import com.telenor.mobileconnect.MobileConnectSdkProfile;
+import com.telenor.mobileconnect.operatordiscovery.OperatorDiscoveryAPI;
```
Comment (gorandostanic) [inline connect/src/com/telenor/connect/AbstractSdkProfile.java]: Done!

File: connect/src/com/telenor/connect/AbstractSdkProfile.java (hunk 5 above)
Comment (gorandostanic) [inline connect/src/com/telenor/connect/AbstractSdkProfile.java]: Done!

File: connect/src/com/telenor/connect/AbstractSdkProfile.java (hunk 4 above)
Comment (gorandostanic) [inline connect/src/com/telenor/connect/AbstractSdkProfile.java]: Done! (renamed to WellKnownConfigStore)

File: connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java (hunk 3 above)
Comment (gorandostanic) [inline connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java]: Done!

File: connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java (hunk 1 above)
Comment (gorandostanic) [inline connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java]: Done!

File: mobileconnect-example/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -14,13 +14,13 @@
         android:name="com.telenor.connect.mobileconnectexample.ExampleApplication"
         android:supportsRtl="true"
         android:theme="@style/AppTheme" >
-        <activity android:name="com.telenor.connect.mobileconnectexample.SignedInActivity" />
-        <activity android:name="com.telenor.connect.mobileconnectexample.SignInActivity">
+        <activity android:name="com.telenor.connect.mobileconnectexample.SignedInActivity">
```
Comment (gorandostanic) [inline mobileconnect-example/src/main/AndroidManifest.xml]: It has changed (now the main activity is SignedInActivity (not SignInActivity as it was before)

File: connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java [hunk 8]
```diff
@@ -247,4 +268,29 @@ public void getUserInfo(
                     userInfoCallback);
         }
     }
+
+    private class OperatorDiscoveryConfigStore {
```
Comment (jorunfa) [inline connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java]: Same as above, I think extracting this class into it's own file would make for better readability.

File: connect/src/com/telenor/connect/AbstractSdkProfile.java [hunk 9]
```diff
@@ -89,4 +99,29 @@ public void failure(RetrofitError error) {
                     }
                 });
     }
+
+    private class WellKnownConfigStore {
```
Comment (gorandostanic) [inline connect/src/com/telenor/connect/AbstractSdkProfile.java]: Done!

File: connect/src/com/telenor/connect/ConnectSdkProfile.java [hunk 10]
```diff
@@ -4,13 +4,17 @@
...
+import com.google.gson.Gson;
 import com.squareup.okhttp.HttpUrl;
 import com.telenor.connect.utils.ConnectUrlHelper;
+import com.telenor.mobileconnect.operatordiscovery.OperatorDiscoveryAPI;
 
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 
+import static com.telenor.connect.utils.ConnectUtils.PREFERENCES_FILE;
```
Comment (gorandostanic) [inline connect/src/com/telenor/connect/ConnectSdkProfile.java]: Done!

File: connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java (hunk 8 above)
Comment (gorandostanic) [inline connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java]: Done!

File: connect/src/com/telenor/connect/WellKnownConfigStore.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    public WellKnownAPI.WellKnownConfig get() {
+        String wellKnownConfigJson = context
+                .getSharedPreferences(PREFERENCES_FILE, Context.MODE_PRIVATE)
+                .getString(PREFERENCE_KEY_WELL_KNOWN_CONFIG, null);
+
+        return preferencesGson.fromJson(
+                wellKnownConfigJson,
+                WellKnownAPI.WellKnownConfig.class);
+    }
+}
```
Comment (jorunfa) [inline connect/src/com/telenor/connect/WellKnownConfigStore.java]: Remember newline at the end.

**Pull request merged**